src/storage-client/src/controller.rs

```diff
@@ -1486,17 +1606,30 @@ where
     #[tracing::instrument(level = "debug", skip(self))]
     fn set_read_policy(&mut self, policies: Vec<(GlobalId, ReadPolicy<Self::Timestamp>)>) {
         let mut read_capability_changes = BTreeMap::default();
+
         for (id, policy) in policies.into_iter() {
```

Comment:
not your code, but `into_iter()` is always implied in for loops, it can be `for (id, policy) in policies {`